pinlockview/src/main/java/com/andrognito/pinlockview/InputField.java

```diff
@@ -0,0 +1,178 @@
+        setRawInputType(InputType.TYPE_CLASS_TEXT);
+        setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                /* do nothing */
+            }
+        });
+        setOnTouchListener(new OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                /* do nothing */
+                return true;
+            }
+        });
+        setOnKeyListener(new OnKeyListener() {
+            @Override
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+                /* do nothing */
+                return true;
+            }
+        });
+        setOnEditorActionListener(new OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                /* do nothing */
+                return true;
+            }
+        });
+    }
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        return true;
+    }
+
+    @Override
+    public boolean onKeyLongPress(int keyCode, KeyEvent event) {
+        return true;
+    }
+
+    @Override
+    public boolean onKeyUp(int keyCode, KeyEvent event) {
+        return true;
+    }
+
+    private void setupPasswordDots() {
+        TextWatcher pinWatcher = new TextWatcher() {
+
+            private Handler pinHandler = new Handler();
+
+            int oldLength = 0;
+            boolean editing = false;
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                if (editing) return; // Do nothing if editing field within the watcher, to prevent infinite loops
+
+                pinHandler.removeCallbacksAndMessages(null);
+
+                Runnable hideText = new Runnable() {
+                    @Override
+                    public void run() {
+                        editing = true;
+                        InputField.this.setText(hideString(InputField.this.getText().toString()));
+                        InputField.this.setSelection(InputField.this.getText().length());
+                        editing = false;
+                    }
+                };
+
+                if (oldLength < s.toString().length()) { // Briefly display last digit if PIN increases
+                    oldLength = s.length();
+
+                    editing = true;
+                    InputField.this.setText(almostHideString(InputField.this.getText().toString()));
+                    InputField.this.setSelection(InputField.this.getText().length());
+                    editing = false;
+
+                    pinHandler.postDelayed(hideText, 1500);
```

Comment:
can we make this configurable? Or at least a class constant so it's easier to find and change?